store/state/value.go

```diff
@@ -0,0 +1,126 @@
+package state
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Value is a capability for reading and writing on a specific key-value point
+// in the state. Value consists of a mapping and a key. An actor holding a Value
+// has a full access right on that state point.
+type Value struct {
```

Comment:
so the `Value` is part of a KVStore.... but the `Value` itself has a map `"key []byte -> value []byte"` and a prefix? This is all super confusing...

Also, why does the value contain the one of the keys of the map?